proposals/exp-show-target.md

```diff
@@ -0,0 +1,214 @@
+information is a priority.
+
+There are several ways in which the included parameters might be irrelevant:
+* Parameters files may include values that aren't tracked by DVC at all (for
+  example, debug level).
+* The parameters may be used in a different pipeline branch (for example, the
+  `topic_modeling` stage is in a separate branch of the pipeline from the
+  `evaluate` stage, which produces the relevant experiment metrics).
+* The parameters may be used in a downstream pipeline branch (for example, the
+  post-processing `combine` stage parameters aren't relevant to experiment
+  evaluation).
+
+```console
+                +-------------------+
+                | data/data.xml.dvc |
+                +-------------------+
+                          *
+                          *
+                          *
+                     +---------+
+                     | prepare |
+                     +---------+
+                          *
+                          *
+                          *
+                    +-----------+
+                    | featurize |
+                  **+-----------+**
+              ****       *         ****
+          ****           *             ***
+        **              *                 ***
++-------+               *                    **
+| train |             **                      *
++-------+            *                        *
+         **        **                         *
+           **    **                           *
+             *  *                             *
+        +----------+                  +----------------+
+        | evaluate |                  | topic_modeling |
+        +----------+                  +----------------+
+                   ***             ****
+                      *         ***
+                       **     **
+                     +---------+
+                     | combine |
+                     +---------+
+```
+
+There are no apparent reasons that users want to see untracked parameters in
+`dvc exp show`. Unlike `dvc params diff` and `dvc metrics diff`, `dvc exp show`
+is not applicable to arbitrary files. Likewise, there are no apparent reasons
+that users want to see parameters or metrics from irrelevant stages of their
+pipelines. Users could keep untracked parameters in other files if they don't
+want them shown in `dvc exp show`, and they could exclude parameters or metrics
+from irrelevant stages, but DVC already has enough information to understand
+that these should not be shown.
+
+# Detailed design
+
+`dvc exp show` could take stages as (positional?) arguments, similar to `dvc exp
+run`. The help message could be similar:
+
+```console
+positional arguments:
+stages               Stages for which to print experiment info. 'dvc.yaml'
+by default.
+```
+
+If no stages are passed, parameters and metrics defined in any stage would be
+shown. This differs from current behavior only because untracked parameters
```

Comment:
> defined in any stage

Any stage in ./dvc.yaml only or any stage anywhere (like `repro -P`?

Reply:
Good point. I think any stage anywhere is better, but we need to work on terminology for this.

Reply:
I think the default targets for `repro` etc is just ./dvc.yaml

Reply:
Ah, thanks for pointing that out. In that case, I guess we should have a default of `./dvc.yaml` and include a `-P, --all-pipelines` option. Seems like maybe I need to go through the repro options and determine all that should apply here.